### PR TITLE
update closure compiler and type fixes (release v20160822)

### PR DESCRIPTION
**Pull request by erwinmombay**
- this fixes some sw.js type errors caused by bad externs typing
- minor type fixes to dom.js

/to @cramforce @jridgewell

### CODE REVIEW

**Comment by cramforce** [comment]
LGTM

**Comment by cramforce** [comment]
Please comment the CC version / github tag

File: src/service-worker/kill.js [hunk 1]
```diff
@@ -20,7 +20,11 @@ self.addEventListener('install', function(event) {
   // Usually when you register a new service worker, it has to wait for a
   // navigation (refreshing the page) before it really starts. Well, we can't
   // wait to kill it, so skip the wait.
-  event.waitUntil(self.skipWaiting());
+  // NOTE(erwinm): add type hint that self is ServiceWorkerGlobalScope until
+  // we can cleanly denote that the global object is ServiceWorkerGlobalScope
+  // and not Window.
+  event.waitUntil(/** @type {!ServiceWorkerGlobalScope} */ (
```
Comment (jridgewell) [inline src/service-worker/kill.js]: Really? What does it think `self` is?

Comment (erwinmombay) [inline src/service-worker/kill.js]: I believe it thinks its `Window`, doesn't know about service worker context. asked @cramforce and there's no easy clean way right now to do it. so just doing it temporarily

**Comment by jridgewell** [comment]
:shipit:

**Comment by erwinmombay** [comment]
added compiler version to PR header and commit message

**Pull request merged**